src/bacnet/bacdef.h

```diff
@@ -126,7 +126,7 @@
 #define BACNET_ARRAY_ALL 0xFFFFFFFFU
 /* For device object property references with no device id defined */
 #define BACNET_NO_DEV_ID   0xFFFFFFFFu
-#define BACNET_NO_DEV_TYPE 0xFFFFu
+#define BACNET_NO_DEV_TYPE ((BACNET_OBJECT_TYPE)0xFFFFu)
```

Comment:
Because BACNET_OBJECT_TYPE is being used as enum everwhere else, this would be better as:
1. add OBJECT_NONE = 0xffffu as last entry into BACNET_OBJECT_TYPE in bacenum.h
2. use OBJECT_NONE for BACNET_NO_DEV_TYPE, or deprecate BACNET_NO_DEV_TYPE.